zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/AQLGenerator.scala

```diff
@@ -102,31 +116,35 @@ class AQLGenerator extends IQLGenerator {
   }
 
   private def parseLookup(lookups: Seq[LookupStatement],
-                          varMap: Map[String, AQLVar]
+                          varMap: Map[String, AQLVar],
+                          schemaMap: Map[String, Schema]
                          ): (String, Map[String, AQLVar]) = {
-    val sb = StringBuilder.newBuilder
     val producedVar = mutable.Map.newBuilder[String, AQLVar]
     lookups.zipWithIndex.foreach { case (lookup, id) =>
       val lookupVar = s"$$l$id"
       val keyZip = lookup.lookupKeys.zip(lookup.sourceKeys).map { case (lookupKey, sourceKey) =>
         varMap.get(sourceKey) match {
-          case Some(aqlVar) => s"$lookupVar.$lookupKey = ${aqlVar.aqlExpr}"
+          case Some(aqlVar) => s"${aqlVar.aqlExpr} /* +indexnl */ = $lookupVar.$lookupKey"
           case None => throw FieldNotFound(sourceKey)
         }
       }
-      sb.append(
+      val returnZip = lookup.as.zip(lookup.selectValues).map { case (asName, selectName) =>
+        s"'$selectName' : $lookupVar.$asName"
+      }
+
+      val subQuery =
         s"""
-           |for $lookupVar in dataset ${lookup.dataset}
+           |(for $lookupVar in dataset ${lookup.dataset}
            |where ${keyZip.mkString(" and ")}
-        """.stripMargin
-      )
-      //TODO check if the vars are duplicated
-      val field: Field = ??? // get field from lookup table
-      producedVar ++= lookup.as.zip(lookup.selectValues).map { p =>
-        p._1 -> AQLVar(new Field(p._1, field.dataType), s"$lookupVar.${p._2}")
-      }
+           |return $lookupVar.${lookup.selectValues.head})[0]
```

Comment:
Can we put a comment for the reason why we only use the first element? Maybe as todos?